typescript/phoenix-sdk/src/utils/connection.ts

```diff
@@ -0,0 +1,125 @@
+import { Connection, PublicKey, SYSVAR_CLOCK_PUBKEY } from "@solana/web3.js";
+import { ZSTDDecoder } from "zstddec";
+
+// Largest configured account size (multiplied by 2)
+const MAX_ACCOUNT_SIZE_BYTES = 1723488 * 2;
+
+/**
+ * Get market account data and clock data with zstd compression
+ * @param marketKey The market address
+ * @returns
+ */
+export const getConfirmedMarketsAndClockAccounts = async (
+  connection: Connection,
+  marketKeys: PublicKey[]
+): Promise<Buffer[]> => {
+  const rawAccounts = (
+    await connection.getMultipleAccountsInfo(
+      [...marketKeys, SYSVAR_CLOCK_PUBKEY],
+      "confirmed"
+    )
+  ).map((account) => account?.data);
+  if (!rawAccounts.every((account) => account !== undefined)) {
+    throw new Error("Unable to get account data");
+  }
+  return rawAccounts as Buffer[];
+};
+
+/**
+ * Get market account data and clock data with zstd compression
+ * @param marketKey The market address
+ * @returns
+ */
+export const getConfirmedMarketsAndClockAccountsZstd = async (
+  connection: Connection,
+  marketKeys: PublicKey[]
+): Promise<Buffer[]> => {
+  const body = {
+    jsonrpc: "2.0",
+    id: 1,
+    method: "getMultipleAccounts",
+    params: [
+      [...marketKeys.map((m) => m.toBase58()), SYSVAR_CLOCK_PUBKEY.toBase58()],
+      {
+        encoding: "base64+zstd",
+        commitment: "confirmed",
```

Comment:
This method and the one following should probably just take an optional CommitmentLevel that defaults to "confirmed"